htdocs/product/class/product.class.php

```diff
@@ -3611,6 +3611,32 @@ public function clone_associations($fromId, $toId)
         return 1;
     }
 
+
+    // phpcs:disable PEAR.NamingConventions.ValidFunctionName.ScopeNotCamelCaps
+    /**
+     *  copy related categories to another product
+     *
+     * @param  int $fromId Id produit source
+     * @param  int $toId   Id produit cible
+     * @return int                 < 0 si erreur, > 0 si ok
+     */
+	function clone_categories($fromId, $toId)
```

Comment:
We try to use now the camelCase notation for all new method we introduce. Can you rename it into 
cloneCategories ?
Also, because cloning categories may be a feature required by all other elements, i would but this method into the class Category instead of product.
You can add a parameter $type='product' so we can replace the ...categorie_product with categorie_'.$type
and have a feature working for all object.

Reply:
another way would be to put the cloneCategories function in the CommonObject class and retrieve the type from $object->element ? it seems a little weird to call $category->cloneCategories, i would prefer $product->cloneCategories, $thidparty->cloneCategories and maybe later $order->cloneCategories and so on.
what do you think of that @eldy ?

Reply:
ping @eldy

Reply:
@eldy  done !

Reply:
@eldy 
i have done the job since April 5th; thanks to merge ;-)